pkg/gui/controllers/files_controller.go

```diff
@@ -748,6 +754,117 @@ func (self *FilesController) createStashMenu() error {
 	})
 }
 
+func (self *FilesController) openCopyMenu() error {
+	node := self.context().GetSelected()
+
+	copyNameItem := &types.MenuItem{
+		Label: self.c.Tr.CopyFileName,
+		OnPress: func() error {
+			name := node.Name()
+			return self.copyToClipboard(name)
+		},
+		Key: 'n',
+	}
+	copyPathItem := &types.MenuItem{
+		Label: self.c.Tr.CopyFilePath,
+		OnPress: func() error {
+			return self.copyToClipboard(node.Path)
+		},
+		Key: 'p',
+	}
+	copyFileDiffItem := &types.MenuItem{
+		Label:   self.c.Tr.CopySelectedDiff,
+		Tooltip: self.c.Tr.CopyFileDiffTooltip,
+		OnPress: func() error {
+			path, hasStaged := self.context().GetSelectedPath(), self.hasPathStagedChanges(node)
+			diff, err := self.c.Git().Diff.GetPathDiff(path, hasStaged)
+			if err != nil {
+				return self.c.Error(err)
+			}
+			return self.copyToClipboard(diff)
+		},
+		Key: 's',
+	}
+	copyAllDiff := &types.MenuItem{
+		Label:   self.c.Tr.CopyAllFilesDiff,
+		Tooltip: self.c.Tr.CopyFileDiffTooltip,
+		OnPress: func() error {
+			diff, err := self.c.Git().Diff.GetAllDiff(self.c.Helpers().WorkingTree.AnyStagedFiles())
+			if err != nil {
+				return self.c.Error(err)
+			}
+			return self.copyToClipboard(diff)
+		},
+		Key: 'a',
+	}
+
+	if node == nil {
+		copyNameItem.DisabledReason = self.c.Tr.FileCopyNoItemSelectedError
+		copyPathItem.DisabledReason = self.c.Tr.FileCopyNoItemSelectedError
+		copyFileDiffItem.DisabledReason = self.c.Tr.FileCopyNoItemSelectedError
+	}
+	if node != nil && !self.anyPathTrackedFile(node) {
+		copyFileDiffItem.DisabledReason = self.c.Tr.NoContentToCopyError
+	}
+	if !self.anyTrackedFile() {
+		copyAllDiff.DisabledReason = self.c.Tr.NoContentToCopyError
+	}
+
+	return self.c.Menu(types.CreateMenuOptions{
+		Title: self.c.Tr.CopyToClipboardMenu,
+		Items: []*types.MenuItem{
+			copyNameItem,
+			copyPathItem,
+			copyFileDiffItem,
+			copyAllDiff,
+		},
+	})
+}
+
+func (self *FilesController) anyPathTrackedFile(node *filetree.FileNode) bool {
+	stagedFiles := node.GetStagedFiles()
+	if len(stagedFiles) == 0 {
+		return node.SomeFile(func(t *models.File) bool {
+			return t.Tracked
+		})
+	}
+
+	for _, file := range stagedFiles {
+		if file.Tracked {
+			return true
+		}
+	}
+	return false
+}
+
+func (self *FilesController) anyTrackedFile() bool {
+	stagedFiles := self.c.Helpers().WorkingTree.GetStagedFiles()
+	if len(stagedFiles) == 0 {
+		return self.c.Helpers().WorkingTree.AnyTrackedFiles()
+	}
+
+	for _, stagedFile := range stagedFiles {
+		if stagedFile.Tracked {
+			return true
+		}
+	}
+	return false
+}
+
+func (self *FilesController) hasPathStagedChanges(node *filetree.FileNode) bool {
+	return node.SomeFile(func(t *models.File) bool {
+		return t.HasStagedChanges
+	})
+}
+
+func (self *FilesController) copyToClipboard(content string) error {
+	if err := self.c.OS().CopyToClipboard(content); err != nil {
+		return self.c.Error(err)
+	}
+	self.c.Toast(self.c.Tr.FileContentCopiedToast)
```

Comment:
OK, I know I argued for a generic text for all items, but that was for the disabled reasons, where I think it's ok. For the toast I find it important to say which item was copied, as a confirmation that you copied the right thing. We also do this for the `y` menu for commits, see [here](https://github.com/jesseduffield/lazygit/blob/d8059d7f7de21bedd2d989f6961f62ec48781463/pkg/i18n/english.go#L1304).